appinventor/components/src/com/google/appinventor/components/runtime/util/JavaStringUtils.java

```diff
@@ -0,0 +1,240 @@
+// -*- mode: java; c-basic-offset: 2; -*-
+// Copyright 2017-2020  MIT, All rights reserve
+// Released under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package com.google.appinventor.components.runtime.util;
+
+import android.util.Log;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Java implementation of string utility methods for use in Scheme calls.
+ * Used for ease of reasoning about the solution or to address issues
+ * with Kawa (e.g. memory problems).
+ * See runtime.scm
+ */
+public class JavaStringUtils {
+  public static final String LOG_TAG_JOIN_STRINGS = "JavaJoinListOfStrings";
+  private static final boolean DEBUG = false;
+
+
+  // Implements the following operation
```

Comment:
@ewpatton Should we move these comments elsewhere? Since the JavaJoinWithStrings class had some loose comments around, I moved them under the method. Let me know if you'd like to re-organize this a bit. Also, I moved the majority to the corresponding Javadoc.